keps/0001a-meta-kep-implementation.md

```diff
@@ -0,0 +1,206 @@
+---
+kep-number: 1a
+title: Meta KEP Implementation
+authors:
+  - "@justaugustus"
+  - "@calebamiles"
+  - "@jdumars"
+owning-sig: sig-pm
+participating-sigs:
+  - sig-architecture
+reviewers:
+  - "@idvoretskyi"
+  - "@timothysc"
+approvers:
+  - "@bgrant0607"
+  - "@jbeda"
+editor:
+creation-date: 2018-09-08
+last-updated: 2018-09-18
+status: provisional
+see-also:
+  - KEP-1
+replaces:
+superseded-by:
+---
+
+# Meta KEP Implementation
+
+## Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Why another KEP?](#why-another-kep)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Implementation Details / Notes / Constraints](#implementation-details--notes--constraints)
+    - [Define](#define)
+    - [Organize](#organize)
+    - [Visibility and Automation](#visibility-and-automation)
+    - [Constraints](#constraints)
+- [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+
+## Summary
+
+Drive KEP adoption through improved process, documentation, visibility, and automation.
+
+## Motivation
+
+The KEP process is the standardized structure for proposing changes to the Kubernetes project.
+
+In order to graduate KEPs to GA, we must iterate over the implementation details.
+
+This KEP seeks to define actionable / delegable items to move the process forward.
+
+Finally, by submitting a KEP, we gain an opportunity to dogfood the process and further identify areas for improvement.
+
+### Why another KEP?
+
+As [KEP-1] is currently de facto for the project, we must be careful to make changes to it in an iterative and atomic fashion.
+
+When proposing a KEP, we action over some unit of work, usually some area of code.
+
+In this instance, we treat [KEP-1] as the unit of work. That said, this would be considered a meta-KEP of the meta-KEP.
+
+### Non-Goals
+
+- API Review process
+- Feature request triage
+- Developer guide
+
+## Proposal
+
+### Implementation Details / Notes / Constraints
+
+#### Define
+
+- Refine existing KEP documentation
+- Define KEP [DACI]
+
+#### Organize
+
+- Move KEPs from flat-files to a directory structure:
+```
+├── 001-introduction.md (required)
+├── 002-motivation.md (required)
+├── 003-goals.md
+├── 004-non-goals.md
+├── 005-guide-for-developers.md (required)
+├── 006-guide-for-teachers.md (required)
+├── 007-guide-for-operators.md (required)
+├── 008-guide-for-project-maintainers.md
+├── 009-graduation-criteria.md (required)
```

Comment:
Maybe this is covered in the videos, but... this is an awful lot of files.  As a reader, am I going to have to keep jumping between files?  How are existing KEP's going to be migrated to this format? I'm confused that the goals/non-goals aren't required.  Are all of these required when I first submit a KEP, or are some of them required to move a KEP from one status to the next?

Reply:
If we're deprecating design-proposals, which of these docs is supposed to describe the proposed design/implementation and alternatives considered?

Reply:
I had the same concern about the about of files. We'll workshop this and get back to you.
It probably makes sense to roll-up the `000..00n` files into one, which includes the currently implemented list:
* [Summary](#summary)
* [Motivation](#motivation)
    * [Goals](#goals)
    * [Non-Goals](#non-goals)
* [Proposal](#proposal)
    * [User Stories [optional]](#user-stories-optional)
      * [Story 1](#story-1)
      * [Story 2](#story-2)
    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
    * [Risks and Mitigations](#risks-and-mitigations)
* [Graduation Criteria](#graduation-criteria)
* [Implementation History](#implementation-history)
* [Drawbacks [optional]](#drawbacks-optional)
* [Alternatives [optional]](#alternatives-optional)

...so maybe no change on that front.

Migration of existing KEPs: haven't nailed that down. Maybe we divide-and-conquer and guide SIGs through the process, similar to what Steering does for SIG Charters?

Goals / Non-goals should be required. Will fix.

> Are all of these required when I first submit a KEP, or are some of them required to move a KEP from one status to the next?

No. We need to define the state machine and entry and exit criteria for each.

> If we're deprecating design-proposals, which of these docs is supposed to describe the proposed design/implementation and alternatives considered?

Should be part of the `Proposal` / `Drawbacks` / `Alternatives` sections, within the KEP.